src/app/core/main.nim

```diff
@@ -25,7 +25,6 @@ proc newStatusFoundation*(fleetConfig: string): StatusFoundation =
   result.signalsManager = newSignalsManager(result.events)
 
 proc delete*(self: StatusFoundation) =
-  self.threadpool.teardown()
```

Comment:
It is generally considered good practice to ensure that all threads are properly terminated before the program exits. I would keep the teardown as follows:
```nim
proc teardown*(self: ThreadPool) =
    sync()
```

see: https://nim-lang.org/docs/threadpool.html#sync

Reply:
Good point! I've seen the threads being properly terminated by default and didn't worry about this.